packages/astro/src/components/HeaderNav.astro

```diff
@@ -1,78 +1,50 @@
 ---
+import NavLink from "./NavLink.astro";
+import NavDropdown from "./NavDropdown.astro";
+
 const routes = [
 	{ label: "Wiki", page: "wiki" },
-    { label: "Get Started", page: "getting-started" },
-    { label: "Events", page: "events" },
-    { label: "Projects", page: "projects" },
-    { label: "Blog", page: "blog" },
-    { label: "Donate", page: "donate" },
+	{ label: "Get Started", page: "getting-started" },
+	{ label: "Events", page: "events" },
+	{ label: "Projects", page: "projects" },
+	{ label: "Blog", page: "blog" },
+	{ label: "Donate", page: "donate" },
 	{ label: "About", page: "about",
-        pages: [
-            { label: "Code of Conduct", page: "about/code-of-conduct" }
-        ]
-    }
+		pages: [
+			{ label: "Code of Conduct", page: "about/code-of-conduct" }
+		]
+	}
 ];
-// because of annoying differences between dev and build modes, due to this
-// behavior (https://github.com/withastro/astro/issues/5630), always remove
-// any trailing slash so currentPage can match the pages above
-const fullPath = Astro.url.pathname.replace(/\/$/, "");
-const currentPage = fullPath.slice(fullPath.lastIndexOf("/") + 1);
 ---
 
-<nav class="container">
-	<ul>
-		<li>
-			<a href="./" class="contrast">
-				<strong>SF Civic Tech</strong>
-			</a>
-		</li>
-	</ul>
-	<ul>
-		{routes.map(route => (
-            <li class={`nav-item ${!!route.pages?.length ? "dropdown" : ""}`}>
-                <a href={route.page} aria-current={currentPage === route.page ? "page" : false}>{route.label}</a>
-                {!!route.pages?.length &&
-                    <div class="dropdown-content">
-                        {route.pages.map(page => (
-                            <a href={page.page} aria-current={currentPage === page.page ? "page" : false}>{page.label}</a>
-                        ))}
-                    </div>
-                }
-            </li>
-        ))}
-	</ul>
-</nav>
+<header class="container">
+	<nav>
+		<ul>
+			<li>
+				{/* the href of "" on the homepage link here looks a little weird, but the base
+					path is prefixed to these links, so basePath + "" == the root page */}
+				<NavLink href="" class="contrast">
+					<strong>SF Civic Tech</strong>
+				</NavLink>
+			</li>
+		</ul>
+		<ul>
+			{routes.map(({ label, page, pages }) => (
+				<li>
+					<NavLink href={page}>
+						{label}
+					</NavLink>
+					{!!pages?.length &&
+						<NavDropdown items={pages} />
```

Comment:
What's the reason we are moving the dropdown to its own component? It seems like its very little code and I don't see it being reused in any other case except for when its used within the navigation bar.

Reply:
Just that it seemed a little cleaner to have all the dropdown CSS in its own file, since all of the styles only applied to that element.  But probably overkill.